### ISSUE TITLE
Duplicate declaration: Ssh_authorized_key[user@domain] is already declared;

### ISSUE THREAD

**Issue by jordigg** [deric/puppet-accounts#40]
Got the following error on 1.2.1:
`Duplicate declaration: Ssh_authorized_key[user1@domain] is already declared;`
This happens when I try to add two SSH Keys with the same name but on different users. Hiera example here:

``` yaml
accounts::users:
  user1:
    comment: "User One"
    ssh_keys:
      user1@domain:
        type: "ssh-rsa"
        key: "AAAAB3NzaC1yc2EAAAADAQABAAABAQC+g9RlVRmA84VPH8+UIszzH6eyRftEBk6cs0YbHRIY9amcWtv41ziCu1shFLRyYdcHpVo0YIrJj+lstpqNVE1PSISc8xYpksKkT81PWH/41YtxXn2VFtNgXGx20ziMXtHQatrs3Ul"
    groups: ['sudo', 'group1']
    shell: '/bin/bash'
  user2:
    comment: "User Two"
    ssh_keys:
      user1@domain:
        type: "ssh-rsa"
        key: "AAAAB3NzaC1yc2EAAAADAQABAAABAQC+g9RlVRmA84VPH8+UIszzH6eyRftEBk6cs0YbHRIY9amcWtv41ziCu1shFLRyYdcHpVo0YIrJj+lstpqNVE1PSISc8xYpksKkT81PWH/41YtxXn2VFtNgXGx20ziMXtHQatrs3Ul"
    groups: ['sudo', 'group1']
    shell: '/bin/bash'
```

Works using a different name for one of the keys or by defining they one single key and adding the name of the key on the comment parameter like:

``` yaml
accounts::users:
  user1:
    comment: "User one"
    ssh_key:
      type: "ssh-rsa"
      key: "AAAAB3NzaC1yc2EAAAADAQABAAABAQC+g9RlVRmA84VPH8+UIszzH6eyRftEBk6cs0YbHRIY9amcWtv41ziCu1shFLRyYdcHpVo0YIrJj+lstpqNVE1PSISc8xYpksKkT81PWH/41YtxXn2VFtNgXGx20ziMXtHQatrs3Ul"
    groups: ['sudo', 'group1']
    shell: '/bin/bash'
```

**Comment by deric** [comment]
Hi Jordi, that's a feature of Puppet :worried:. There's the [ticket](https://tickets.puppetlabs.com/browse/PUP-2621), the problem is there since Puppet 0.24, I guess we can't expect that it would be fixed any time soon.

Yes, you're right. For single `ssh_key` it should work. The naming pattern is `"${username}_${ssh_key['type']}"`. I guess we could do the same `ssh_keys` - it would just duplicate all keys on existing installations (unless `purge_ssh_keys => true` is enabled). 

I guess in next major release we could deprecate `ssh_key` parameter and prefix all `ssh_keys` with username, just to have clearer API.

**Comment by jordigg** [comment]
Ups sorry about that! Though it could be related to the other open issue about "duplicate declarations" but it isn't.
I think what you say makes sense and I vote to do the same. Shouldn't be a big deal having repeated public keys. It won't be an issue on new setups and would make the API way cleaner with only one single way to add keys. `ssh_key` should be deprecated in favour of `ssh_keys`.
I already changed my configs to use ssh_keys but got that error on one specific case so I wanted to check with you to see if the newer versions fixed that problem. I guess we may need to wait for Puppetlabs.
Thanks Deric!

**Comment by deric** [comment]
Resolved in [1.6 branch](https://github.com/deric/puppet-accounts/tree/1.6).
